frontend/utils/translation.ts

```diff
@@ -13,23 +16,37 @@ export function formatMultiPartSpan (t: ComposerTranslation, key: string, classe
   return `<span>${parts.join(' ')}</span>`
 }
 
-// Translation with default. Needed if we want to default to '' as the translation lib does not support to default to ''
+/** Translation with default. Needed if we want to default to '' as the translation lib does not support to default to '' */
 export function tD (t: ComposerTranslation, path : string, options?: any, d:string = '') : string {
-  const translation = t(path, 'NOT_FOUND', { ...options, missingWarn: false })
-  return (translation === 'NOT_FOUND') ? d : translation
+  if (typeof options === 'number') {
+    options = { plural: options }
+  }
+  const translation = t(path, NOT_FOUND, { ...options, missingWarn: false })
+  return (translation === NOT_FOUND) ? d : translation
 }
 
-// get a translation at a specific index, if it exists
+function tOfWithEOA (t: ComposerTranslation, path : string, index : number, options?: any) : string {
+  return tD(t, `${path}[${index}]`, options, END_OF_ARRAY)
+}
+
+/** gets a translation at a specific index, if it exists */
 export function tOf (t: ComposerTranslation, path : string, index : number, options?: any) : string {
-  return tD(t, `${path}[${index}]`, options)
+  const element = tOfWithEOA(t, path, index, options)
+  return (element === END_OF_ARRAY) ? '' : element
 }
 
-// get all translations for an array
+/** returns an array of translations (with only one element if the translation data is a simple string) */
 export function tAll (t: ComposerTranslation, path : string, options?: any) : string[] {
   const list: string[] = []
-  let index = 0
-  for (let value = tOf(t, path, index); value; value = tOf(t, path, ++index, options)) {
-    list.push(value)
+  if (tD(t, path, options, NOT_FOUND) !== NOT_FOUND) {
```

Comment:
```suggestion
  if (tD(t, path, option)) {
```
tD returns '' if not found so this check should be sufficient

Reply:
The problem is that `''` caused `tAll()` to stop when a translation part was empty (which might happen for example because in some language this part is at a different place (like a verb in German that moves to the end)).
So I made this change to make sure that `tAll()` really stops at the end of the array and never before.

Reply:
You are right, we would return an empty array, but I do not find consistent to return `[]` (meaning "no translation found") when there is a translation (which an empty string but still, the field in the JSON is not missing so I find it more consistent to return `[""]`).